code/modules/games/cards.dm

```diff
@@ -0,0 +1,472 @@
+
+/obj/item/weapon/pack
+	name = "Card Pack"
+	desc = "For those with disposable income."
+
+	icon_state = "card_pack"
+	icon = 'icons/obj/playing_cards.dmi'
+	w_class = WEIGHT_CLASS_TINY
+	var/list/cards = list()
+	var/parentdeck = null // For future card pack that need to be compatible with eachother i.e. cardemon
+
+
+/obj/item/weapon/pack/attack_self(mob/user as mob)
+	user.visible_message("<span class = 'notice'>[name] rips open the [src]!</span>")
+	var/obj/item/weapon/cardhand/H = new()
+
+	H.cards += cards
+	cards.Cut();
+	user.drop_item()
+	qdel(src)
+
+	H.update_icon()
+	user.put_in_active_hand(H)
+
+/obj/item/weapon/cardhand
+	name = "hand of cards"
+	desc = "Some playing cards."
+	icon = 'icons/obj/playing_cards.dmi'
+	icon_state = "empty"
+	w_class = WEIGHT_CLASS_TINY
+	throw_speed = 4
+	throw_range = 20
+	throwforce = 0
+	force = 0
+	var/maxcardlen = 20
+
+	var/concealed = FALSE
+	var/list/cards = list()
+	var/parentdeck = null
+
+/obj/item/weapon/cardhand/attackby(obj/O as obj, mob/user as mob)
+	if(cards.len == 1 && istype(O, /obj/item/weapon/pen))
+		var/datum/playingcard/P = cards[1]
+		if(P.name != "Blank Card")
+			to_chat(user,"<span class = 'notice'>You cannot write on that card.</span>")
+			return
+		var/cardtext = sanitize(input(user, "What do you wish to write on the card?", "Card Editing") as text|null, MAX_PAPER_MESSAGE_LEN)
+		if(!cardtext)
+			return
+		P.name = cardtext
+		// SNOWFLAKE FOR CAG, REMOVE IF OTHER CARDS ARE ADDED THAT USE THIS.
+		P.card_icon = "cag_white_card"
+		update_icon()
+	else if(istype(O,/obj/item/weapon/cardhand))
+		var/obj/item/weapon/cardhand/H = O
+		if((H.cards.len + cards.len) > maxcardlen)
+			to_chat(user,"<span class = 'warning'>You can't hold that many cards in one hand!</span>")
+			return
+		if(H.parentdeck == src.parentdeck)
+			for(var/datum/playingcard/P in cards)
+				H.cards += P
+			H.concealed = src.concealed
+			qdel(src)
+			H.update_icon()
+			return
+		else
+			to_chat(user,"<span class = 'notice'>You cannot mix cards from other deck!</span>")
+			return
+	..()
+
+/obj/item/weapon/cardhand/attack_self(var/mob/user as mob)
+	concealed = !concealed
+	update_icon()
+	user.visible_message("[user] [concealed ? "conceals" : "reveals"] their hand.")
+
+/obj/item/weapon/cardhand/examine(mob/user)
+	..(user)
+	if((!concealed) && cards.len)
+		to_chat(user,"<span class = 'notice'>It contains:</span class>")
+		for(var/datum/playingcard/P in cards)
+			to_chat(user,"<span class = 'notice'>the [P.name].</span class>")
+
+/obj/item/weapon/cardhand/verb/Removecard()
+
+	set category = "Object"
+	set name = "Remove card"
+	set desc = "Remove a card from the hand."
+	set src in view(1)
+
+	var/mob/living/carbon/user = usr
+
+	if(user.incapacitated() || !Adjacent(user)) return
+
+	var/pickablecards = list()
+	for(var/datum/playingcard/P in cards)
+		pickablecards[P.name] += P
+	var/pickedcard = input("Which card do you want to remove from the hand?")	as null|anything in pickablecards
```

Comment:
Errant tab in here, it should be `(space)as null|anything`

Reply:
I'm a bit confused about this. What does an Errant tab mean?